### ISSUE TITLE
Match exhaustiveness check

### ISSUE THREAD

**Issue by csi-adziahel** [atomicobject/monadt#1]
The case when not all cases are matched should be addressed, with an exception (Haskell-style) or a warning (OCaml-style). /me prefers Haskell and optional wrapper that catches an exception and issues a warning.

**Comment by ShalokShalom** [comment]
I agree on that and ask myself, if this library is still actually usable now?
